noobaa/noobaa-core
# replace proxy-agent with https-proxy-agent

### Explain the changes:
- replace proxy-agent with https-proxy-agent
  - we used `proxy-agent` as an agent for connections through a proxy.  
  - proxy-agent used `https-proxy-agent@1.0.0`  which had a bug where it didn't propagate `rejectUnauthorized: false` to `tls.connect` call (see https://github.com/TooTallNate/node-https-proxy-agent/pull/30)
  - this bug is fixed in newer versions of `https-proxy-agent`, so instead of using `proxy-agent` to create a generic agent, I changed to create an agent according to the endpoint protocol.
- Upgrade behind proxy 
  - ignore NTP test if a proxy is configured and system works on local time
- added cloud resource alerts


### Issues: Fixed / Gap #xxx
- Fixes #4474 
- Fixes #4476
- Fixes #4477 

### Testing Instructions:
- in the proxy environment, run a proxy according to these instructions:
```
I managed to reproduce it with the following proxy. Please triple check me.

Proxy used: https://github.com/joeferner/node-http-mitm-proxy

I installed it on the proxy server in your lab (10.1.16.7). Installed under /tmp/mitm

run the proxy with node node-proxy-mitm.js

it will open port 8081
```

### Reminders:
- Create a new test - the first is the hardest
- User Experience is top priority
- Design for failure
- Keep it simple
- `npm test`
- Imagine a support call - Add diagnostics info, phonehome info, activity log events, external syslog
- Upgrade - DB schema, server platform, agents install, npm packages
